exonum/src/storage/proof_map_index/key.rs

```diff
@@ -131,82 +178,84 @@ impl DBKey {
         }
     }
 
-    /// Shortens this DBKey to the specified length.
-    pub fn prefix(&self, suffix: u16) -> DBKey {
-        debug_assert!(self.from + suffix <= self.data.len() as u16 * 8);
+    fn start_from(&self, start: u16) -> Self {
+        debug_assert!(start <= self.end());
 
-        DBKey {
-            data: self.data,
-            from: self.from,
-            to: self.from + suffix,
-        }
+        let mut key = DBKey::from_raw(self.bytes);
+        key.start = start;
+        key
     }
 
-    /// Return object which represents a view on to this slice (further) offset by `i` bits.
-    pub fn suffix(&self, suffix: u16) -> DBKey {
-        debug_assert!(self.from + suffix <= self.to);
+    fn prefix(&self, pos: u16) -> Self {
+        debug_assert!(self.start() + pos <= self.raw_key().len() as u16 * 8);
 
-        DBKey {
-            data: self.data,
-            from: self.from + suffix,
-            to: self.to,
-        }
+        let mut key = DBKey::from_raw(self.bytes);
+        key.start = self.start;
+        key.set_end(self.start + pos);
+        key
     }
 
-    /// Shortens this DBKey to the specified length.
-    pub fn truncate(&self, size: u16) -> DBKey {
-        debug_assert!(self.from + size <= self.to);
+    fn suffix(&self, pos: u16) -> Self {
+        debug_assert!(self.start() + pos <= self.end());
```

Comment:
The difference between `start_from` and `suffix` is pretty subtle :( Do we really need both of this methods? Should we implement `suffix` in terms of `start_from` as `fn suffix(&self, pos: u16) -> Self { self.start_from(self.start() + pos) }`